base/utfconvert.jl

```diff
@@ -0,0 +1,546 @@
+# This file is a part of Julia. License is MIT: http://julialang.org/license
+
+# Functions to convert to different UTF encodings
+
+# Quickly copy and set trailing \0
+@inline function fast_utf_copy(T::Type{UInt16}, len, dat)
+    @inbounds return UTF16String(setindex!(copy!(Vector{T}(len), dat), 0, len))
+end
+@inline function fast_utf_copy(T::Type{Char}, len, dat)
+    @inbounds return UTF32String(setindex!(copy!(Vector{T}(len), dat), 0, len))
+end
+
+# Get rest of character ch from 3-byte UTF-8 sequence in str
+@inline function get_utf8_3(dat, pos, ch)
+    @inbounds return ((ch & 0xf) << 12) | (UInt32(dat[pos-1] & 0x3f) << 6) | (dat[pos] & 0x3f)
+end
+
+# Get rest of character ch from 4-byte UTF-8 sequence in dat, update pos and return character
+@inline function get_utf8_4(dat, pos, ch)
```

Comment:
This should be `get_utf8_4!` if it's mutating... otherwise the comment is out of date.

Same for `get_utf8_3!` and `output_utf8_4!`, the latter is definitely mutating. (Not sure how you lost `get_utf8_2` and other friends...)